### DISCUSSION TITLE
If velero support backup CRD and CR together then restore with the right sequence?

### DISCUSSION THREAD

**Discussion by joshua-bj** [vmware-tanzu/velero#6930] [Community support Q&A]
I tried to backup the CRD and related CR at the same velero backup, when I restore the backup, the velero report issue for the CR like 
`time="2023-10-09T06:34:46Z" level=info msg="Skipping restore of resource because it cannot be resolved via discovery" logSource="/remote-source/velero/app/pkg/restore/restore.go:1833" resource=certificates.cert-manager.io restore=openshift-adp/restore-cert-manager`

According for the issue #2948, it should be supported, while I still can find this issue at version:
```
velero version -n openshift-adp
Client:
        Version: v1.9.5
        Git commit: 2b5281f38aad2527f95b55644b20fb169a6702a7
Server:
        Version: v1.9.5-OADP
```
```
oc version
Client Version: 4.11.18
Kustomize Version: v4.5.4
Server Version: 4.13.4
Kubernetes Version: v1.26.5+7d22122
```

**Comment by sseago** [comment]
Velero restores CRDs first, then refreshes to avoid exactly this error. Can you verify that:
1) backup includes the CRD
2) restore did restore the CRD? If so, before or after the failed attempt to restore the CR?

**Comment by joshua-bj** [comment]
Thanks @sseago , I made a mistake here, I thought the CRD has been included in the backup while it was not. It's still strange why the CRD is there at the restored OCP cluster. I'm sure about I deleted them before restore the system,  and the CRD is not at the `<backup-name>-resource-list.json`. Will velero install the CRD automatically when install CR if the catalog source is ready?

The good news is after I include the CRD at the backup, everything is fine. Very appreciate for your quick turn around.
